extension/zpagesextension/zpagesextension.go

```diff
@@ -24,8 +25,9 @@ type zpagesExtension struct {
 	config              *Config
 	telemetry           component.TelemetrySettings
 	zpagesSpanProcessor *zpages.SpanProcessor
-	server              http.Server
+	server              *http.Server
 	stopCh              chan struct{}
+	httpSettings        confighttp.HTTPServerSettings
```

Comment:
that's not quite right, this should be in the Config struct.

Reply:
can you elaborate more over this

Reply:
Here is an example of use: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/healthcheckextension/config.go#L28
Here is how it's used https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/healthcheckextension/healthcheckextension.go#L35-L41

Reply:
I have added it to the Config struct and have taken care for the types of the parameters being passed by the methods but dont know why am failing over this test which is related to difference between the Endpoint found and the given in the test.

Test Failure log :

```
config_test.go:32: 
                Error Trace:    /media/sanket-teli/X1/opentelemetry-collector/extension/zpagesextension/config_test.go:32
                Error:          Not equal: 
                                expected: &zpagesextension.Config{HTTPServerSettings:confighttp.HTTPServerSettings{Endpoint:"", TLSSetting:(*configtls.TLSServerSetting)(nil), CORS:(*confighttp.CORSSettings)(nil), Auth:(*configauth.Authentication)(nil), MaxRequestBodySize:0, IncludeMetadata:false, ResponseHeaders:map[string]configopaque.String(nil)}, TCPAddr:confignet.TCPAddr{Endpoint:"localhost:56888", DialerConfig:confignet.DialerConfig{Timeout:0}}}
                                actual  : &zpagesextension.Config{HTTPServerSettings:confighttp.HTTPServerSettings{Endpoint:"localhost:56888", TLSSetting:(*configtls.TLSServerSetting)(nil), CORS:(*confighttp.CORSSettings)(nil), Auth:(*configauth.Authentication)(nil), MaxRequestBodySize:0, IncludeMetadata:false, ResponseHeaders:map[string]configopaque.String(nil)}, TCPAddr:confignet.TCPAddr{Endpoint:"localhost:56888", DialerConfig:confignet.DialerConfig{Timeout:0}}}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -2,3 +2,3 @@
                                  HTTPServerSettings: (confighttp.HTTPServerSettings) {
                                -  Endpoint: (string) "",
                                +  Endpoint: (string) (len=15) "localhost:56888",
                                   TLSSetting: (*configtls.TLSServerSetting)(<nil>),
                Test:           TestUnmarshalConfig
2024/01/27 04:48:37 http: panic serving 127.0.0.1:42552: runtime error: invalid memory address or nil pointer dereference

```